packages/shared-integration/src/constants.ts

```diff
@@ -5,4 +5,5 @@ export const CSS_PLACEHOLDER = '@unocss-placeholder'
 export const SKIP_START_COMMENT = '@unocss-skip-start'
 export const SKIP_END_COMMENT = '@unocss-skip-end'
 // eslint-disable-next-line regexp/no-super-linear-backtracking, regexp/no-useless-lazy, regexp/optimal-quantifier-concatenation
-export const SKIP_COMMENT_RE = new RegExp(`(\/\/\\s*?${SKIP_START_COMMENT}\\s*?|\\/\\*\\s*?${SKIP_START_COMMENT}\\s*?\\*\\/|<!--\\s*?${SKIP_START_COMMENT}\\s*?-->)[\\s\\S]*?(\/\/\\s*?${SKIP_END_COMMENT}\\s*?|\\/\\*\\s*?${SKIP_END_COMMENT}\\s*?\\*\\/|<!--\\s*?${SKIP_END_COMMENT}\\s*?-->)`, 'g')
+export const SKIP_UNO_COMMENT_RE = new RegExp(`(\/\/\\s*?${SKIP_START_COMMENT}\\s*?|\\/\\*\\s*?${SKIP_START_COMMENT}\\s*?\\*\\/|<!--\\s*?${SKIP_START_COMMENT}\\s*?-->)[\\s\\S]*?(\/\/\\s*?${SKIP_END_COMMENT}\\s*?|\\/\\*\\s*?${SKIP_END_COMMENT}\\s*?\\*\\/|<!--\\s*?${SKIP_END_COMMENT}\\s*?-->)`, 'g')
+export const SKIP_COMMENT_RE = /^\s*(<!--.*?-->|\/\/.*)/gm
```

Comment:
This would create a lot of false-negative, RegExp is not turning complete - removing comments correctly would require a proper parser - where UnoCSS is designed to be language-agnosic (+parsing would cost a lot computational power as well as bundle size), we shouldn't do this.

In the end, Uno's approach is trying to sacrifice a bit false-positive (which is just extra necessary bits) to avoid false-negative (which would cause user apps to break unexpectedly). I believe this is the best workaround so far. For the most accuracy, I'd say users can provide custom extractors for each language and use a proper parser to extract - but that would be out of scope of Uno I'd say.

In that case I would suggest to not have this PR.

Reply:
It suddenly occurred to me that this implementation of skipping comments can be implemented by users themselves in transformers